employee-rostering-frontend/src/domain/DomainObjectView.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import DomainObject from './DomainObject';
+
+ type DomainObjectView<T extends DomainObject> = {
+   [K in keyof T]: 
+   T[K] extends DomainObject[]? number[] :
+     T[K] extends DomainObject? number :
+       T[K] extends (DomainObject | null)? number | null :
+         T[K];
+ }
+ 
+/* eslint-disable no-undef */
+export default DomainObjectView;
+/* eslint-enable no-undef */
```

Comment:
If you want to disable a rule on only one line you can use `// eslint-disable-next-line no-undef` before that line or `// eslint-disable-line no-undef` on the same line.